unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -0,0 +1,216 @@
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+  GIVEN("A generic java type and a concrete substitution")
+  {
+    symbol_tablet symbol_table;
+
+    generate_java_generic_typet type_generator(message_handler);
+
+    auto boxed_type_reference=symbol_typet("java.lang.Integer");
+    auto boxed_parameter=java_generic_inst_parametert(boxed_type_reference);
+    auto list_type_reference=java_type_from_string("Ljava/util/List;");
+    java_generic_typet generic_type(list_type_reference);
+    generic_type.generic_type_variables().push_back(boxed_parameter);
+
+    struct_typet base_class;
+    base_class.components().emplace_back(
+      struct_union_typet::componentt(
+        "a",
+        java_type_from_string("TT;", "com.diffblue.Foo")));
+    base_class.set_tag("UserClass");
+
+    generic_type.subtype()=base_class;
+
+    const symbolt &new_symbol=type_generator(generic_type, symbol_table);
+    REQUIRE(new_symbol.is_type);
+    REQUIRE(new_symbol.base_name=="UserClass<java.lang.Integer>");
+    REQUIRE(new_symbol.type.id()==ID_struct);
+  }
+
+}
+
+SCENARIO(
+  "generate_java_generic_type_from_file",
+  "[CORE][java_bytecode][generate_java_generic_type]")
+{
+  std::unique_ptr<languaget> java_lang(new_java_bytecode_language());
+
+  // Configure the path loading
+  cmdlinet command_line;
+  command_line.set(
+    "java-cp-include-files",
+    "./java_bytecode/generate_concrete_generic_type");
+  config.java.classpath.emplace_back(
+    "./java_bytecode/generate_concrete_generic_type");
+
+  // Configure the language
+  null_message_handlert message_handler;
+  java_lang->get_language_options(command_line);
+  java_lang->set_message_handler(message_handler);
+
+  std::istringstream java_code_stream("ignored");
+
+
+  GIVEN("A generic java type with two generic fields and a concrete "
+          "substitution")
+  {
+    java_lang->parse(java_code_stream, "generic_two_fields.class");
+
+    symbol_tablet new_symbol_table;
+    java_lang->typecheck(new_symbol_table, "");
+
+    java_lang->final(new_symbol_table);
+
+    REQUIRE(new_symbol_table.has_symbol(
+      "java::generic_two_fields$bound_element<java::java.lang.Integer>"));
+    THEN("The class should contain two instantiated fields.")
+    {
+      const symbolt &class_symbol=new_symbol_table.lookup(
+        "java::generic_two_fields$bound_element<java::java.lang.Integer>");
+      const typet &symbol_type=class_symbol.type;
+
+      REQUIRE(symbol_type.id()==ID_struct);
+      class_typet class_type=to_class_type(symbol_type);
+      REQUIRE(class_type.is_class());
+
+      auto generic_components_counter=0;
```

Comment:
I think this would be better to explicitly get the components:

```
REQUIRE(`class_type.has_component("first"));
const auto &first_component=class_type.get_component("first");
REQUIRE(is_java_generic_parameter(first_component.type()));
REQUIRE(`class_type.has_component("second"));
const auto &second_component=class_type.get_component("second");
REQUIRE(is_java_generic_parameter(second_component.type()));
```
Or even pull it out into a function that verifies that the relevant component exists without duplication.

Reply:
This for loop can just be deleted right?